### PR TITLE
Upgrade to Apple Silicon on CI, simplify Apple Silicon installation instructions

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Upgrade the macOS CI test to run on Apple Silicon. Also simplify Apple Silicon installation instructions a bit.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Review by nilsdeppe — DISMISSED** [review]
Thank you!

**Comment by knelli2** [comment]
MacOS spack installation of pybind11 isn't happy

File: .github/workflows/Tests.yaml
Comment (knelli2) [inline .github/workflows/Tests.yaml]: The `SetupPybind11.cmake` isn't finding the boostrapped pybind11. Maybe we need another hint in cmake?

**Review by knelli2 — CHANGES_REQUESTED** [review]

**Comment by nilsvu** [comment]
@geoffrey4444 FYI. Still needs #5956 to be merged first.

**Comment by geoffrey4444** [comment]
I'm excited about adding Apple Silicon to the CI...thanks for doing this!

I understand there are complexities to #5956 , so if that looks like it won't get merged for a while, maybe consider just implementing the slightly longer instructions in a separate PR, so we can start automatically checking Apple Silicon? We can always improve things later, but we have enough Apple Silicon users that it would be nice to get the CI up and running. But if $5956 is going to get merged soon-ish, maybe a separate PR isn't worth it.

For those who prefer manual installation (I'm probably one of them, at least for now), maybe instead of deleting the longer instructions, move them down to a "manually install dependencies" section?

**Comment by nilsvu** [comment]
@geoffrey4444 ok I removed the dependence on #5956 for now. Once it's merged we can simplify the instructions further, but I agree that it would be good to keep an (optional) "install manually" step.

File: docs/Installation/InstallationOnAppleSilicon.md
Comment (geoffrey4444) [inline docs/Installation/InstallationOnAppleSilicon.md]: Why no pybind11 here? Is this taken care of by step 2 above (i.e., the requirements or dev_requirements)?

File: .github/workflows/Tests.yaml
Comment (geoffrey4444) [inline .github/workflows/Tests.yaml:875]: Can you please make 2 macOS tests: i) Apple Silicon, and ii) Intel?

**Review by geoffrey4444 — CHANGES_REQUESTED** [review]
Thanks again for doing this, @nilsvu ! I reviewed the Apple-specific commits (the last two). Is it possible to just spin them off into a separate PR, to get the Apple Silicon docker container merged ASAP? The other changes don't seem to affect the Apple Silicon instructions or adding Apple Silicon to the docker container.

File: docs/Installation/InstallationOnAppleSilicon.md
Comment (nilsvu) [reply docs/Installation/InstallationOnAppleSilicon.md]: yes, that's one of the earlier commits in this PR.

File: .github/workflows/Tests.yaml
Comment (nilsvu) [reply .github/workflows/Tests.yaml:875]: Oh I don't think we need Intel anymore, do we?

Comment (nilsdeppe) [reply .github/workflows/Tests.yaml:875]: People still use Intel Macs. They aren't that old yet so I think we should continue to test them

**Comment by nilsvu** [comment]
@nilsdeppe @geoffrey4444 @knelli2 I rebased and updated this. Please, could whoever of you is interested in this take another look so we can get this in? It would be helpful to have an ARM macOS build on CI.

File: docs/Installation/InstallationOnAppleSilicon.md
Comment (knelli2) [inline docs/Installation/InstallationOnAppleSilicon.md:164]: I guess it's up to @geoffrey4444 if he wants the docs default to install extra dependencies in people's build directories

**Review by knelli2 — COMMENTED** [review]
Changes look ok to me, but I don't have a mac to test.

**Review by nilsdeppe — DISMISSED** [review]

File: docs/Installation/InstallationOnAppleSilicon.md
Comment (geoffrey4444) [reply docs/Installation/InstallationOnAppleSilicon.md:164]: I think fetching missing dependencies is fine

**Comment by geoffrey4444** [comment]
LGTM, but I notice it has conflicts that must be resolved first

**Comment by nilsvu** [comment]
Rebased!

**Review by knelli2 — DISMISSED** [review]

**Comment by knelli2** [comment]
Tolerances on the partial derivative test still seem to not be adequate.

**Comment by nilsdeppe** [comment]
No need for my approval on this. I was pinged in the past, but have nothing to add.

**Comment by nilsvu** [comment]
Tolerances seem to work now @knelli2

**Review by knelli2 — APPROVED** [review]

**Pull request merged**
